src/main/java/org/broadinstitute/hellbender/tools/copynumber/CreateBinningIntervals.java

```diff
@@ -0,0 +1,122 @@
+package org.broadinstitute.hellbender.tools.copynumber;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.VariantProgramGroup;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.File;
+import java.util.List;
+
+
+
+@CommandLineProgramProperties(
+        summary = "Split intervals into sub-interval files.",
+        oneLineSummary = "Split intervals into sub-interval files.",
+        programGroup = VariantProgramGroup.class
+)
+@DocumentedFeature
+public class CreateBinningIntervals extends GATKTool {
+    public static final String WIDTH_OF_BINS_SHORT_NAME = "bw";
+    public static final String WIDTH_OF_BINS_LONG_NAME = "binwidths";
+
+    public static final String PADDING_SHORT_NAME = "pad";
+    public static final String PADDING_LONG_NAME = "padding";
+
+    @Argument(
+            doc = "width of the bins",
+            fullName = WIDTH_OF_BINS_LONG_NAME,
+            shortName = WIDTH_OF_BINS_SHORT_NAME,
+            optional = true,
+            minValue = 1
+    )
+    private int widthOfBins = 1;
+
+    @Argument(
+            doc = "width of the padding regions",
+            fullName = PADDING_LONG_NAME,
+            shortName = PADDING_SHORT_NAME,
+            optional = true,
+            minValue = 0
+    )
+    private int padding = 0;
+
+    @Argument(
+            doc = "output file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    public File outputFile;
+
+
+
+    @Override
+    public void onTraversalStart() {
+        createBins();
+    }
+
+    /**
+     * Generates binning coverage in the intervals given by the user.
+     * The width of bins, the intervals and the output file's path are given by the user.
+     */
+    public void createBins() {
+        // check if the output directory exists
+        if (!outputFile.exists() && !outputFile.mkdir()) {
+            throw new RuntimeException("Unable to create file: " + outputFile.getAbsolutePath());
```

Comment:
This should be an `UserException`, I guess... and maybe it will be better to provide this in  `customCommandLineValidation()` or `onStartup()` (if the rest of the logic is moved to `doWork()`.

Reply:
I ended up removing all of my tests (since the positivity of padding and the length of bins is done automatically and, as @samuelklee pointed out, I shouldn't check for the existence of the file). 
Therefore, I didn't need to overwrite `customCommandLineValidation()`. But I will use it in the future.